State.lua

```diff
@@ -1393,60 +1392,64 @@ do
         return b == nil or ( a.next < b.next )
     end
 
-
     -- Increase max forecast duration because Assassination is pooling hard this tier.
     local FORECAST_DURATION = 10.01
 
-    forecastResources = function( resource )
-
+    forecastResources = function ( resource )
         if not resource then return end
 
+        -- Initialize or wipe tables
+        events = events or {}
+        remains = remains or {}
         wipe( events )
         wipe( remains )
 
-        local now = state.now + state.offset -- roundDown( state.now + state.offset, 2 )
-
-        local timeout = FORECAST_DURATION * state.haste -- roundDown( FORECAST_DURATION * state.haste, 2 )
+        local now = state.now + state.offset
+        local timeout = FORECAST_DURATION * state.haste
 
         if state.class.file == "DEATHKNIGHT" and state.runes then
-            timeout = max( timeout, 0.01 + 2 * state.runes.cooldown )
+            -- Adjust timeout based on rune cooldowns and regen models for Frost DK
+            timeout = max( timeout, 0.01 + state.runes.cooldown * ( 6 - state.runes.current ) )
```

Comment:
This, in theory, can put the timeout at 60 seconds with 0 haste and you somehow spent all 6 runes at once.  Forecasting beyond 15 seconds doesn't actually accomplish anything.  I think this would short circuit upon reaching max runes anyway, but realistically, I think this keeps it constrained to useful values:

`timeout = max( timeout, 0.01 + state.runes.expiry[ 6 ] - query_time )`

That is assuming `expiry` is still sorted (and resorted) such that actual runes are reordered based on when they come back up.

Reply:
Yep, they are sorted.